Source/Plugins/CoherenceViewer/CoherenceNode.h

```diff
@@ -83,25 +83,27 @@ class CoherenceNode : public GenericProcessor, public Thread
     
 
 private:
-    // group of 3, controlled by coherenceSync:
 
     AtomicallyShared<Array<FFTWArray>> dataBuffer;
     AtomicallyShared<std::vector<std::vector<double>>> meanCoherence;
 
+    // we should refactor so that these only exist during acquisition
+    // so that updateDataBufferSize and maybe other functions can be called to
+    // modify the underlying data when settings are changed.
```

Comment:
To elaborate, the write and read pointers are really meant to be constructed temporarily at the point where they're used and then destroyed. It might be a little more efficient to keep them around while acquisition is active, but then they should be destroyed in `disable` (or when the thread exits) so that the `AtomicallyShared` objects know there are no readers or writers and it's safe to use `apply` to modify the objects.

Reply:
Sounds good!